src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackReader.cs

```diff
@@ -626,29 +626,49 @@ public unsafe double ReadDouble()
         /// <returns>The value.</returns>
         public DateTime ReadDateTime(ExtensionHeader header)
         {
-            if (header.TypeCode != ReservedMessagePackExtensionTypeCode.DateTime)
+            if (header.TypeCode != ReservedMessagePackExtensionTypeCode.DateTime &&
+                header.TypeCode != ReservedMessagePackExtensionTypeCode.DateTimeFluentForward)
```

Comment:
I don't think extensions defined externally to the msgpack spec should be supported directly by the `MessagePackReader` type. Instead I would expect to support this through an `IMessagePackFormatter`. Such a class would not even have to be defined in this library, but if it is sufficiently popular then we can certainly consider it.